packages/server/src/automations/loopUtils.ts

```diff
@@ -18,14 +18,18 @@ function replaceBindingsRecursive(
   value: string | ObjValue,
   loopStepNumber: number
 ) {
+  if (value === null || value === undefined) {
+    return value
+  }
+
   if (typeof value === "object") {
     for (const [innerKey, innerValue] of Object.entries(value)) {
       if (typeof innerValue === "string") {
         value[innerKey] = automationUtils.substituteLoopStep(
           innerValue,
           `steps.${loopStepNumber}`
         )
-      } else if (typeof innerValue === "object") {
+      } else if (typeof innerValue === "object" && innerValue !== null) {
```

Comment:
I know this isn't your code but `typeof innerValue === "object"` almost certainly doesn't do what the author is expecting it to. 

```
❯ node
Welcome to Node.js v20.10.0.
Type ".help" for more information.
> typeof {} === "object"
true
> typeof [] === "object"
true
> typeof null === "object"
true
```

Reply:
I think it is mine from a good while ago - but yes you're right. We would be better checking for the existence of keys in the object here I think

Reply:
UPD2: I added some more checks in that if, in order to be a bit more clear on what we're doing.